packages/dx-react-core/src/utils/with-components.tsx

```diff
@@ -1,7 +1,9 @@
 import * as React from 'react';
 
-const makeBoundComponent = (Target, components, exposed) => {
+const makeBoundComponent = (Target, components, exposed: object) => {
   class Component extends React.PureComponent {
+    static components: {[key: string]: string};
```

Comment:
Declare an alias for `{ [key: string]: string }` type and use it here and in `ITargetComponent`.